extensions/ql-vscode/src/view/data-extensions-editor/LibraryRow.tsx

```diff
@@ -119,11 +120,11 @@ export const LibraryRow = ({
 
   const handleModelWithAI = useCallback(
     async (e: React.MouseEvent) => {
-      onGenerateFromLlmClick(externalApiUsages, modeledMethods);
+      onGenerateFromLlmClick(title, externalApiUsages, modeledMethods);
```

Comment:
Does this work correctly in framework mode? In that case the `title` is the package name and there is no "dependency".

Reply:
Yeah it seems to work.

Perhaps the name I'm using is a bit wrong? What do we call these things if not "dependency"?

Reply:
Sorry I wasn't very clear. The variable does have a value, but I was worried it might confuse the LLM so wanted to make sure you were aware the value might not be what you expect.

> Perhaps the name I'm using is a bit wrong? What do we call these things if not "dependency"?

Maybe "source" or something like that? In [`sorting.ts`](extensions/ql-vscode/src/data-extensions-editor/shared/sorting.ts) we call them "groupings". I'm not sure either of those are better though.

Alternatively if there is not sensible value it could contain in framework mode then maybe make it optional?

Reply:
Ah, I see, thanks for the clarification. I think it's still okay - we essentially need a unique enough name to manage automodeling jobs so that seems like it would work. I think the value is exactly what we want - it's the name of the package. It's just that the name is confusing, but `source` and `groupings` both sound more confusing to me 😄 is it perhaps a `package`?

Reply:
Since we don't have a clear name right away and I've got some changes stacked up on top of this one, I might go ahead with `dependency` for now and circle back to rename later.

Reply:
Sounds good to me